python/pyarrow/tests/test_dataset.py

```diff
@@ -569,6 +570,22 @@ def test_partitioning():
         with pytest.raises(pa.ArrowInvalid):
             partitioning.parse(shouldfail)
 
+    partitioning = ds.FilenamePartitioning(
+        pa.schema([
+            pa.field('group', pa.int64()),
+            pa.field('key', pa.float64())
+        ])
+    )
+    assert partitioning.dictionaries is None
```

Comment:
Are dictionaries tested somewhere else?

Reply:
I think we have some dictionaries check like this for Hive Partitioning,
https://github.com/apache/arrow/blob/68e48cbf808806eacd7439c25fd116111d291b94/python/pyarrow/tests/test_dataset.py#L3280
Should we have something similar for FilenamePartitioning as well?

Reply:
It would be nice IMHO.

Reply:
Added the test. I noticed that the `dictionaries()` method might have an issue where it returned `None` object even if the Partitioning object has a dictionary field.

Thanks to @westonpace for finding the cause and suggesting the fix.

Reply:
@sanjibansg is there a specific reason you changed this to return an empty list instead of None? 

I am not necessarily saying that the original API to return None was better, but it _is_ a change in behaviour that we should be a bit conscious about since there might already be people relying on this (and so only do this if there is a good reason for it). 

You mentioned

>  the dictionaries() method might have an issue where it returned None object even if the Partitioning object has a dictionary field.

but I don't directly see any additional case in the tests related to this?

Reply:
Ah, so it's for the case where only a subset of your fields would be dictionary encoded, I see. 

Now, in that case returning a list with shorter length can also a bit confusing: to get the dictionaries for a certain partition key, you would need to count how many dictionary encoded columns are present in the schema before the specific key you are looking for.  
Another option could be to always return a list of the same length as the number of fields in the schema, but then with `None` entries for keys that are not dictionary encoded?  (eg `[None, pa.array(["first", "second", "third"])]` for the specific case in the tests)

Reply:
I opened https://issues.apache.org/jira/browse/ARROW-16113 to follow-up on this specific aspect

Reply:
Sure, let me assign it to myself and I will make a PR. We can have the change where list of same length will always be returned with `None` for fields not dictionary encoded.